lib/controller/controller.go

```diff
@@ -1106,3 +1110,31 @@ func (ctrl *ProvisionController) getStorageClassFields(name string) (string, map
 func claimToClaimKey(claim *v1.PersistentVolumeClaim) string {
 	return fmt.Sprintf("%s/%s", claim.Namespace, claim.Name)
 }
+
+func (ctrl *ProvisionController) fetchReclaimPolicy(storageClassName string) (v1.PersistentVolumeReclaimPolicy, error) {
+	classObj, found, err := ctrl.classes.GetByKey(storageClassName)
+	if err != nil {
+		return "", err
+	}
+	if !found {
+		return "", fmt.Errorf("StorageClass %q not found", storageClassName)
+	}
+
+	var reclaimPolicy v1.PersistentVolumeReclaimPolicy
+	switch class := classObj.(type) {
+	case *storage.StorageClass:
+		reclaimPolicy = *class.ReclaimPolicy
+	case *storagebeta.StorageClass:
+		reclaimPolicy = *class.ReclaimPolicy
+	default:
+		return reclaimPolicy, fmt.Errorf("Cannot convert object to StorageClass: %+v", classObj)
+	}
+
+	// At present, the reclaim policy for dynamic provisioned volume only support Retain or Delete.
+	if reclaimPolicy != v1.PersistentVolumeReclaimRetain && reclaimPolicy != v1.PersistentVolumeReclaimDelete {
```

Comment:
it is kubernetes's job to make sure this never happens, we can safely remove it

Reply:
sure, for current controller_test, it's only testing the controller's behavior for v1.5.0, I'd like also to add a test for reclaimPolicy for v1.8.0

Reply:
that sgtm